.teamcity/settings.kts

```diff
@@ -74,4 +79,333 @@ object Chocolatey : BuildType({
             }
         }
     }
+})
+
+object ChocolateyDockerWin : BuildType({
+    id = AbsoluteId("ChocolateyDockerWin")
+    name = "Docker (Windows)"
+
+    params {
+        // TeamCity has suggested "${Chocolatey.depParamRefs.buildNumber}"
+        param("env.CHOCOLATEY_VERSION", "%dep.Chocolatey.build.number%")
+    }
+
+    vcs {
+        root(DslContext.settingsRoot)
+    }
+
+    steps {
+        step {
+            name = "Login Docker"
+            conditions {
+                exists("system.DockerUsername")
+                exists("system.DockerPassword")
+            }
+            type = "DockerLogin"
+        }
+
+        dockerCommand {
+            name = "Build Docker Image"
+            commandType = build {
+                source = file {
+                    path = "docker/Dockerfile.windows"
+                }
+                contextDir = "."
+                platform = DockerCommandStep.ImagePlatform.Windows
+                namesAndTags = "chocolatey/choco:latest-windows"
+            }
+        }
+
+        // powerShell {
+        //     name = "Test Docker Image"
+        //     scriptMode = script {
+        //         content = """
+        //             docker run --rm chocolatey/choco:latest-windows choco.exe --version
+        //             exit ${'$'}LastExitCode
+        //         """.trimIndent()
+        //     }
+        // }
+
+        dockerCommand {
+            name = "Push Docker Image"
+            conditions {
+                exists("system.DockerUsername")
+                exists("system.DockerPassword")
+            }
+            commandType = push {
+                namesAndTags = "chocolatey/choco:latest-windows"
+                removeImageAfterPush = false
+            }
+        }
+
+        dockerCommand {
+            name = "Tag Docker Image with Version"
+            commandType = other {
+                subCommand = "tag"
+                commandArgs = "chocolatey/choco:latest-windows chocolatey/choco:v%env.CHOCOLATEY_VERSION%-windows"
+            }
+        }
+
+        dockerCommand {
+            name = "Push Versioned Tag"
+            conditions {
+                exists("system.DockerUsername")
+                exists("system.DockerPassword")
+            }
+            commandType = push {
+                namesAndTags = "chocolatey/choco:v%env.CHOCOLATEY_VERSION%-windows"
+            }
+        }
```

Comment:
@JPRuskin just having a look through this, and I am curious if there is a reason that this wasn't added as Cake tasks, rather than direct steps within the TeamCity build configuration?

While this build is the only one currently that needs it, there is a good chance that other builds will want/need the ability to build a docker image of the output from the build.  Having this functionality within something like Chocolatey.Cake.Recipe, behind a feature flag (or on the presence of a Docker file in a known location), would mean that we could then deploy it across all build configuraitons.

Also, putting it in the Cake script would mean that we could run the Docker steps locally, if/when required.